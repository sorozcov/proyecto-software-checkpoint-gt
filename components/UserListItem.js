import React, { Component } from 'react';
import { connect } from 'react-redux';
import { ListItem, Left, Thumbnail, Body, Text, Right, Button } from 'native-base';

import default_pic from '../src/resources/default.png';
import * as actionsUsers from '../src/actions/users';


class UserListItem extends Component{
    constructor(props){
        super();
        this.image = (props.image!=null ? `https://firebasestorage.googleapis.com/v0/b/software-checkpoint-gt.appspot.com/o/UserImages%2F${props.image}_400x400.jpg?alt=media` : null);
        this.style = props.style;
    }
    render(){
     
        return(
            <ListItem thumbnail style={{...this.style}}>
                <Left>
                    <Thumbnail circle source={this.image === null ? default_pic : {uri: this.image}} />
                </Left>
                <Body>
                    <Text style={{fontFamily:'dosis-semi-bold',fontSize:18}}>{this.props.name}</Text>
                    <Text style={{fontFamily:'dosis-semi-bold',fontSize:15}} note numberOfLines={1}>{this.props.description}</Text>
                </Body>
               
            </ListItem>
        );
    }
}

export default connect(
    undefined,
    dispatch => ({
      selectUser(navigation, user) {
        dispatch(actionsUsers.selectUser(user));
        navigation.navigate('EditUserScreen');
      },
    }),
  )(UserListItem);
  