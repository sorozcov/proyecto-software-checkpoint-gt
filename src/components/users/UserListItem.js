import { Body, Left, ListItem, Text, Thumbnail } from 'native-base';
import React, { Component } from 'react';
import default_pic from '../../assets/resources/default.png';



class UserListItem extends Component{
    constructor(props){
        super();
        this.image = (props.image != null ? `https://firebasestorage.googleapis.com/v0/b/software-checkpoint-gt.appspot.com/o/UserImages%2F${props.image}_400x400.jpg?alt=media` : null);
        this.style = props.style;
    }

    render(){ 
        return(
            <ListItem thumbnail style={{...this.style}}>
                <Left>
                    <Thumbnail circle source={this.image === null ? default_pic : {uri: this.image, cache:'force-cache'}} />
                </Left>
                <Body>
                    <Text style={{ fontFamily: 'dosis-semi-bold', fontSize: 18 }}>{this.props.name}</Text>
                    <Text style={{ fontFamily:'dosis-semi-bold', fontSize: 15 }} note numberOfLines={1}>{this.props.description}</Text>
                </Body>
               
            </ListItem>
        );
    }
}

export default UserListItem;