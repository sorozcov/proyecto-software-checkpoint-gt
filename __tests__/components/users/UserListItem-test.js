import React from 'react';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import { render } from 'react-native-testing-library';

import UserListItem from '../../../src/components/users/UserListItem';


const mockStore = configureStore([]);
const store = mockStore({
})

describe("Snapshot", () => {
    beforeEach(() => {
        jest.spyOn(console, 'error').mockImplementation(() => {});
    });

    it('User List Item snapshot', () => {
        const component = render(
            <Provider store={store}><UserListItem /></Provider>
        );
    
        let tree = component.toJSON();
    
        expect(tree).toMatchSnapshot();
    })
    
})
