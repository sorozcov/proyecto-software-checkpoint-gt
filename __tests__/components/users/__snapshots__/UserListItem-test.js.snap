// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot User List Item snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#c9c9c9",
          "flexDirection": "row",
          "marginLeft": 16,
          "paddingBottom": null,
          "paddingRight": null,
          "paddingTop": null,
        },
        Object {},
      ]
    }
    thumbnail={true}
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "center",
          "flex": 0,
          "flexDirection": "row",
        }
      }
    >
      <Image
        circle={true}
        source={1}
        style={
          Object {
            "borderRadius": 28,
            "height": 56,
            "width": 56,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": null,
          "alignSelf": null,
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flex": 1,
          "marginLeft": 15,
          "paddingBottom": 18,
          "paddingTop": 18,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
              "marginLeft": null,
              "marginRight": 10,
            },
            Object {
              "fontFamily": "dosis-semi-bold",
              "fontSize": 18,
            },
          ]
        }
        uppercase={false}
      />
      <Text
        note={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "200",
              "marginLeft": null,
              "marginRight": 10,
            },
            Object {
              "fontFamily": "dosis-semi-bold",
              "fontSize": 15,
            },
          ]
        }
        uppercase={false}
      />
    </View>
  </View>
</View>
`;
